R/bundlePackageRenv.R

```diff
@@ -34,26 +34,19 @@ parseRenvDependencies <- function(bundleDir, snapshot = FALSE) {
     return(data.frame())
   }
 
-  if (snapshot) {
-    # Can use system libraries
-    deps$description <- lapply(deps$Package, package_record)
-  } else {
-    old <- options(renv.verbose = FALSE)
-    defer(options(old))
-
-    # Generate a library from the lockfile
-    lib_dir <- dirCreate(file.path(bundleDir, "renv_library"))
-    renv::restore(bundleDir, library = lib_dir, prompt = FALSE)
-    defer(unlink(lib_dir, recursive = TRUE))
-
-    deps$description <- lapply(
-      deps$Package,
-      package_record,
-      # Ensure we fall back to system libraries
-      lib_dir = c(lib_dir, .libPaths())
-    )
+  if (!snapshot) {
+    lib_versions <- unlist(lapply(deps$Package, packageDescription, fields = "Version"))
```

Comment:
You are about to load all the package descriptions just below; what about validating against `deps$description` rather than loading the same information twice?

Reply:
Doh